homeassistant/components/tcp/common.py

```diff
@@ -0,0 +1,161 @@
+"""Common code for TCP component."""
+from __future__ import annotations
+
+import logging
+import select
+import socket
+import ssl
+from typing import Final
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA as PARENT_PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PAYLOAD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_TIMEOUT,
+    CONF_UNIT_OF_MEASUREMENT,
+    CONF_VALUE_TEMPLATE,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import TemplateError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.template import Template
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_BUFFER_SIZE,
+    CONF_VALUE_ON,
+    DEFAULT_BUFFER_SIZE,
+    DEFAULT_NAME,
+    DEFAULT_SSL,
+    DEFAULT_TIMEOUT,
+    DEFAULT_VERIFY_SSL,
+)
+from .model import TcpSensorConfig
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+TCP_PLATFORM_SCHEMA: Final = PARENT_PLATFORM_SCHEMA.extend(
```

Comment:
The common schema should be used to extend each platform parent schema in the platforms.

Reply:
sounds logical, will do so :+1: